imagetest/test_suites/storage_perf/storage_perf_utils.go

```diff
@@ -0,0 +1,123 @@
+	hyperdiskSize = 100
+	bootdiskSize  = 50
+	mountDiskName = "hyperdisk"
+	// TODO: Set up constants for compute.Disk.ProvisionedIOPS int64, and compute.Disk.ProvisionedThrougput int64, then set these fields in appendCreateDisksStep
+)
+
+// FIOOutput defines the output from the fio command
+type FIOOutput struct {
+	Jobs []FIOJob               `json:"jobs,omitempty"`
+	X    map[string]interface{} `json:"-"`
+}
+
+// FIOJob defines one of the jobs listed in the FIO output.
+type FIOJob struct {
+	ReadResult  FIOStatistics          `json:"read,omitempty"`
+	WriteResult FIOStatistics          `json:"write,omitempty"`
+	X           map[string]interface{} `json:"-"`
+}
+
+// FIOStatistics give information about FIO performance.
+type FIOStatistics struct {
+	IOPS      float64                `json:iops,omitempty"`
+	Bandwidth float64                `json:bw_mean,omitempty"`
+	X         map[string]interface{} `json:"-"`
+}
+
+// BlockDeviceList gives full information about blockdevices, from the output of lsblk.
+type BlockDeviceList struct {
+	BlockDevices []BlockDevice `json:"blockdevices,omitempty"`
+}
+
+// BlockDevice defines information about a single partition or disk in the output of lsblk.
+type BlockDevice struct {
+	Name string `json:"name,omitempty"`
+	Size string `json:"size,omitempty"`
+	Type string `json:"type,omitempty"`
+	// Other fields are not currently used.
+	X map[string]interface{} `json:"-"`
+}
+
+// This method currently only runs the commands for linux. Support for getting the partition on windows will be added in the future.
+func getMountDiskPartition(diskExpectedSizeGb int) (string, error) {
+	lsblkCmd := "lsblk"
+	if !utils.CheckLinuxCmdExists(lsblkCmd) {
+		return "", fmt.Errorf("could not find lsblk")
+	}
+	lsblkout, err := exec.Command(lsblkCmd, "-o", "name,size,type", "--json").CombinedOutput()
+	if err != nil {
+		return "", fmt.Errorf("failed to execute lsblk cmd with error: %v", err)
+	}
+
+	var blockDevices BlockDeviceList
+	if err := json.Unmarshal(lsblkout, &blockDevices); err != nil {
+		return "", fmt.Errorf("failed to unmarshal lsblk output with error: %v", err)
+	}
+
+	diskExpectedSizeGbString := strconv.Itoa(diskExpectedSizeGb) + "G"
+	for _, blockDev := range blockDevices.BlockDevices {
+		if strings.ToLower(blockDev.Type) == "disk" && blockDev.Size == diskExpectedSizeGbString {
+			return blockDev.Name, nil
+		}
+	}
+
+	return "", fmt.Errorf("disk block with size not found")
+}
+
+// installFio tries to install fio with any of multiple package managers, and returns an error if all the package managers were not found or failed.
+func installFio() error {
+	var updateCmd, installFioCmd *exec.Cmd
+	if utils.CheckLinuxCmdExists("apt") {
+		updateCmd = exec.Command("apt", "-y", "update")
+		installFioCmd = exec.Command("apt", "install", "-y", "fio")
+	} else if utils.CheckLinuxCmdExists("yum") {
+		updateCmd = exec.Command("yum", "update")
```

Comment:
Probably avoid doing this- it may take a long time. This updates the entire system and isn't necessary.

Reply:
done